### PR TITLE
[oceanbase] use real charset to convert change record

### LINKED ISSUE #2114: [Bug] ob-oracle用flinkcdc抽取nvarchar字段乱码
### Search before asking

- [X] I searched in the [issues](https://github.com/ververica/flink-cdc-connectors/issues) and found nothing similar.

### Flink version

版本：flink-connector-oceanbase-cdc-2.3.0 / OB 3.2.2
截屏2023-05-06 15 19 19

### Flink CDC version

flink-connector-oceanbase-cdc-2.3.0

### Database and its version

3.2.2

### Minimal reproduce step

insert nvarchar 字段中文

### What did you expect to see?

乱码

### What did you see instead?

。。。

### Anything else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by whhe**
fix #2114

### CODE REVIEW

File: flink-connector-oceanbase-cdc/src/main/java/com/ververica/cdc/connectors/oceanbase/table/OceanBaseRecord.java [hunk 1]
```diff
@@ -32,8 +31,8 @@ public class OceanBaseRecord implements Serializable {
     private final boolean isSnapshotRecord;
     private Map<String, Object> jdbcFields;
     private DataMessage.Record.Type opt;
-    private Map<String, ByteString> logMessageFieldsBefore;
-    private Map<String, ByteString> logMessageFieldsAfter;
+    private Map<String, Object> logMessageFieldsBefore;
+    private Map<String, Object> logMessageFieldsAfter;
```
Comment (ruanhang1993) [inline flink-connector-oceanbase-cdc/src/main/java/com/ververica/cdc/connectors/oceanbase/table/OceanBaseRecord.java:35]: final

**Review by ruanhang1993 — COMMENTED** [review]
Thanks for contribution, @whhe . Have these changes been covered by the existed tests?

File: flink-connector-oceanbase-cdc/src/main/java/com/ververica/cdc/connectors/oceanbase/table/OceanBaseRecord.java (hunk 1 above)
Comment (whhe) [reply flink-connector-oceanbase-cdc/src/main/java/com/ververica/cdc/connectors/oceanbase/table/OceanBaseRecord.java]: Fixed.

**Comment by whhe** [comment]
> Thanks for contribution, @whhe . Have these changes been covered by the existed tests?

Yes.

**Review by ruanhang1993 — APPROVED** [review]
LGTM.

**Pull request merged**